File: frontend/src/types/robot-cell.ts
import { CellState } from './enums/validation-status';

export interface RobotCell {
  id: number;
  cell_code: string;
  name: string;
  layout_geojson: GeoJSONValue;
  robot_model: string;
  controller_model: string;
  max_reach_mm: number;
  owner_team: string;
  cell_state: CellState;
  layout_version: number;
  zone_count: number;
  program_count: number;
  created_at: string;
  updated_at: string;
}

export type GeoJSONValue = Record<string, unknown>;

export interface CreateRobotCell {
  cell_code: string;
  name: string;
  layout_geojson: GeoJSONValue;
  robot_model: string;
  controller_model: string;
  max_reach_mm: number;
  owner_team: string;
}

export interface UpdateRobotCell extends Omit<CreateRobotCell, 'cell_code'> {
  layout_version: number;
}

File: backend/internal/model/robot_cell.go
package model

import "time"

type RobotCell struct {
	ID              uint      `gorm:"primaryKey"`
	CellCode        string    `gorm:"size:64;uniqueIndex;not null"`
	Name            string    `gorm:"size:160;not null"`
	LayoutGeoJSON   string    `gorm:"type:text;not null"`
	RobotModel      string    `gorm:"size:120;not null"`
	ControllerModel string    `gorm:"size:120;not null"`
	MaxReachMM      float64   `gorm:"not null"`
	OwnerTeam       string    `gorm:"size:120;index;not null"`
	CellState       string    `gorm:"size:24;index;not null"`
	LayoutVersion   int       `gorm:"not null;default:1"`
	CreatedBy       uint      `gorm:"index;not null"`
	CreatedAt       time.Time `gorm:"not null"`
	UpdatedAt       time.Time `gorm:"not null"`
}
